flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/table/DynamoDbDynamicSink.java

```diff
@@ -0,0 +1,198 @@
+        this.overwriteByPartitionKeys = overwriteByPartitionKeys;
+    }
+
+    @Override
+    public ChangelogMode getChangelogMode(ChangelogMode changelogMode) {
+        // TODO: We can support consuming from CDC streams here
+        return ChangelogMode.insertOnly();
+    }
+
+    @Override
+    public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
+        DynamoDbSinkBuilder<RowData> builder =
+                DynamoDbSink.<RowData>builder()
+                        .setDestinationTableName(destinationTableName)
+                        .setFailOnError(failOnError)
+                        .setOverwriteByPartitionKeys(new ArrayList<>(overwriteByPartitionKeys))
+                        .setDynamoDbProperties(dynamoDbClientProperties)
+                        .setElementConverter(new RowDataElementConverter(physicalDataType));
+
+        addAsyncOptionsToSinkBuilder(builder);
+
+        // TODO: check if parallelism needed here
+        return SinkV2Provider.of(builder.build());
+    }
+
+    @Override
+    public DynamicTableSink copy() {
+        return new DynamoDbDynamicSink(
+                maxBatchSize,
+                maxInFlightRequests,
+                maxBufferedRequests,
+                maxBufferSizeInBytes,
+                maxTimeInBufferMS,
+                destinationTableName,
+                failOnError,
+                dynamoDbClientProperties,
+                physicalDataType,
+                overwriteByPartitionKeys);
+    }
+
+    @Override
+    public String asSummaryString() {
+        // TODO: check when this is called/returned
+        return "DynamoDB";
+    }
+
+    @Override
+    public void applyStaticPartition(Map<String, String> partitions) {
+        this.overwriteByPartitionKeys.addAll(partitions.keySet());
+    }
+
+    public static DynamoDbDynamicTableSinkBuilder builder() {
+        return new DynamoDbDynamicTableSinkBuilder();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
+            return false;
+        }
+        DynamoDbDynamicSink that = (DynamoDbDynamicSink) o;
+        return failOnError == that.failOnError
+                && Objects.equals(destinationTableName, that.destinationTableName)
+                && Objects.equals(dynamoDbClientProperties, that.dynamoDbClientProperties)
+                && Objects.equals(physicalDataType, that.physicalDataType)
+                && Objects.equals(overwriteByPartitionKeys, that.overwriteByPartitionKeys);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
+                super.hashCode(),
+                destinationTableName,
+                failOnError,
+                dynamoDbClientProperties,
+                physicalDataType,
+                overwriteByPartitionKeys);
+    }
```

Comment:
Why are these needed? Please review https://flink.apache.org/contributing/code-style-and-quality-java.html#equals--hashcode and remove if we can

Reply:
Yep, these were only used for tests. Removed and improved tests.